pinot-common/src/main/java/org/apache/pinot/common/utils/UploadedRealtimeSegmentName.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils;
+
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+/**
+ * Class to represent segment names like: uploaded_{tableName}_{partitionId}_{sequenceId}_{creationTime}
+ *
+ * This naming convention is adopted to represent a batch generated segment uploaded to a realtime table. The naming
+ * convention has been kept different from {@LLCSegmentName} to differentiate between batch generated segments and
+ * low level consumer segments.
+ */
+public class UploadedRealtimeSegmentName implements Comparable<UploadedRealtimeSegmentName> {
+
+  public static final String UPLOADED_REALTIME_SEGMENT_NAME_REGEX = "^uploaded_(.+)_(\\d+)_(\\d+)_(\\d+)$";
```

Comment:
hmm. why not just split the fields out from the segment name via StringUtils.split()? In case the table name might include `_`, we can get the required fields as counted from the end of name. And for simplicity, we can enforce to have `seqId` field (which seems optional in this regex pattern).

Reply:
The regex helped avoid parsing numbers during validation. Using separator based approch becomes more complex with handling number parsing.

Regex `uploaded_(.+)_(\\d+)_(\\d+)_(\\d+)$` is handling sequenceId as well.

Reply:
As to the naming pattern, I wonder it may be more flexible to change `uploaded` as an optional suffix and not enforce the suffix value.

e.g. `{tableName}_{partId}_{seqId}_{ctime}_{optional suffix}`. e.g. I may want some minion tasks to upload segments but to differentiate from those uploaded manually, I may use the task name as the suffix. wdyt?

Reply:
I did a quick benchmarking with 10 million strings and String.split() method is takes 0.57 amount of time compared to regex based. Sounds good to go with String.split() based approach.

Test 1
```
Method 1 (String.split()) took: 2253611542 ns
Method 2 (regex match) took: 3891154500 ns
Method 2 is 0.5791626988853822 times faster than Method 1
```

Test 2
```
Method 1 (String.split()) took: 2156181167 ns
Method 2 (regex match) took: 3929024375 ns
Method 2 is 0.5487828430690253 times faster than Method 1
```

Test 3
```
Method 1 (String.split()) took: 2167354959 ns
Method 2 (regex match) took: 3646892125 ns
Method 2 is 0.5943019109730316 times faster than Method 1
```